### ISSUE TITLE
Backup pools

### ISSUE THREAD

**Issue by kiwina** [LostSoulFly/MinerProxy#11]
It be great to add backup pools

**Comment by LostSoulFly** [comment]
Yes, I have had a few problems with my pool going down as well. This is
very high on my todo list but I'm busy working on another project at the
moment.

**Comment by LostSoulFly** [comment]
I haven't released a new version yet, but I wrote a simple failover update: b481d09

[Here's the preview update compiled](https://github.com/LostSoulFly/MinerProxy/files/1252368/MinerProxy.zip) if you or anyone else is unable to compile it themselves.

You can use an existing JSON with it and it should add/update missing variables. Or you can start with a fresh/new one but just running it.

**Comment by kiwina** [comment]
Great already using it but haven't had a chance to see if it actually switches

**Comment by LostSoulFly** [comment]
I've had it work. Switched me to a different pool automatically, then I checked and saw my rigs weren't connected in my pool rig list. I may add an option to switch back to the main pool at some point.

**Comment by hogze94** [comment]
There's something not quite right with this, I've tried 1 pool and the same pool entered twice but for some reason it doesn't recover after internet issues. I dont have logging enabled and i cant check but i find that just restarting the proxy will be enough, the miners dont need to be restarted.

It seems like the proxy drops the connection to the pool and cant recover.

last hour: 234 shares submitted
two hours ago: 2464 shares submitted
three hours ago: 5364 shares submitted

**Comment by LostSoulFly** [comment]
A new connection is made to the pool for every new miner, the pool failover
changes if the last few attempts didn't succeed.

**Comment by hogze94** [comment]
Interesting that the logs dont have the miner name, it just says NONE. And error 10024 is related to sockets


> WSAEMFILE10024 | Too many open files.Too many open sockets. Each implementation may have a maximum number of socket handles available, either globally, per process, or per thread.
> -- | --

**Comment by LostSoulFly** [comment]
Try restarting your computer -- that looks like a pretty low level windows
socket error. I've never encountered that message, and the few searches I
just did was unhelpful. I've got several MinerProxy instances running with
anywhere from 1 to 100 connections simultaneously, sometimes chaining
between one MinerProxy to another at a remote location (intricate web of
redundancy) and I've not had any issues. If I had to guess, I'd say this
issue is machine specific. If I find anything else out about this I'll
reply to this thread, but more information is needed before we can begin to
solve this.
